lib/compilers/java.js

```diff
@@ -103,6 +103,29 @@ export class JavaCompiler extends BaseCompiler {
         ];
     }
 
+    async handleInterpreting(key, executeParameters) {
+        executeParameters.args = [
+            '-Xss136K', // Reduce thread stack size
+            '-XX:CICompilerCount=2', // Reduce JIT compilation threads. 2 is minimum
+            '-XX:-UseDynamicNumberOfCompilerThreads',
+            '-XX:-UseDynamicNumberOfGCThreads',
+            '-XX:+UseSerialGC', // Disable parallell/concurrent garbage collector
```

Comment:
These are all options used to reduce the memory and processes/threads used by the JVM.